helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/AssignmentMetadataStore.java

```diff
@@ -52,6 +52,7 @@ public class AssignmentMetadataStore {
   protected volatile Map<String, ResourceAssignment> _globalBaseline;
   protected volatile Map<String, ResourceAssignment> _bestPossibleAssignment;
   protected volatile int _bestPossibleVersion = 0;
+  protected volatile int _lastPersistedBestPossibleVersion = 0;
```

Comment:
Can we initialize these two variables with the ZK state i.e. path:
```/ASSIGNMENT_METADATA/BEST_POSSIBLE/LAST_SUCCESSFUL_WRITE```

Reply:
In order to do this, we would have to add a public method to BucketDataAccessor interface. This will cause a backwards incompatible change.

Let's hold off on this for now, as the actual version does not matter in the context of the AssignmentMetadataStore object. The use of version here is local to the controller thread processing the pipeline for the cluster. We only need to maintain version here to make sure we flush the cache to ZK eventually.